cli/src/pcluster/validators/slurm_settings_validator.py

```diff
@@ -10,40 +10,86 @@
 # limitations under the License.
 
 from enum import Enum
+from typing import Dict, List
 
 from pcluster.validators.common import FailureLevel, Validator
 
 # SLURM SETTINGS are case-insensitive - keep them lowercase since they are compared with setting.lower()
 SLURM_SETTINGS_DENY_LIST = {
-    "Queue": ["nodes", "partitionname", "resumetimeout", "state", "suspendtime"],
-    "ComputeResource": ["cpus", "features", "gres", "nodeaddr", "nodehostname", "nodename", "state"],
+    "SlurmConf": {
+        "Global": [
+            "slurmctldhost",
+            "suspendtime",
+            "resumetimeout",
+            "slurmuser",
+            "reconfigflags",
+            "launchparameters",
+            "slurmctldparameters",
+            "treewidth",
+            "communicationparameters",
+            "suspendprogram",
+            "resumeprogram",
+            "resumefailprogram",
+            "grestypes",
+            "slurmctldlogfile",
+            "slurmdlogfile",
+            "taskplugin",
+            "jobcomppass",
```

Comment:
[Nitpick] Keeping these values in alphabetical order will make it easier to maintain the list

Reply:
Addressing in the next commit.